dvc/ignore.py

```diff
@@ -340,7 +350,11 @@ def check_ignore(self, target):
     def is_ignored(self, path):
         # NOTE: can't use self.check_ignore(path).match for now, see
         # https://github.com/iterative/dvc/issues/4555
-        return self.is_ignored_dir(path) or self.is_ignored_file(path)
+        if os.path.isfile(path):
```

Comment:
I see us using `os.path` in a few other places instead of using the `self.tree`. Might be a potential bug in non-local trees (e.g. GitTree).

Reply:
Also, these two `ifs` are clearly an optimization, right? For use in `OutputBase` or elsewhere?

Reply:
> I see us using `os.path` in a few other places instead of using the `self.tree`. Might be a potential bug in non-local trees (e.g. GitTree). Need to double check that, this PR might worsen it.

Yes, but the question is that our `self.tree.isfile` and `self.tree.isdir` rely on `dvcignore` itself. If we use `self.tree` here, they would become circular dependent.

Reply:
> Also, these two `ifs` are clearly an optimization, right? For use in `OutputBase` or elsewhere?

The original one is a bug, a file might match a dir-only pattern.

Reply:
@efiop @karajan1001, maybe we need to get back to using `use_dvcignore` flag?